### PR TITLE
DEVPROD-8373 Update spruce and parsley analytics attributes to use dot notation

### PR DESCRIPTION
**Pull request by khelif96**
DEVPROD-8373

### Description
This PR updates all of the remaining spruce and parsley analytics attributes to utilize dot notation which is recommended/required by OTEL. Also removes the usage of complex structs since those are not supported. 
In a future PR I will introduce strict typescript enforcement of structs so that they are not complex.

Sorry for the large number of files changed the changes are small.

**Have you have updated the analytics documentation if necessary?**  
Yes

### CODE REVIEW

File: apps/spruce/src/pages/spawn/spawnVolume/spawnVolumeTableActions/MigrateVolumeModal.tsx [hunk 1]
```diff
@@ -107,13 +106,10 @@ export const MigrateVolumeModal: React.FC<MigrateVolumeModalProps> = ({
...
     sendEvent({
       name: "Created a spawn host",
-      isMigration: true,
-      // @ts-expect-error: FIXME. This comment was added by an automated script.
-      params: omit(mutationInput, [
-        "publicKey",
-        "userDataScript",
-        "setUpScript",
-      ]),
+      "is.volume.migration": true,
```
Comment (SupaJoon) [inline apps/spruce/src/pages/spawn/spawnVolume/spawnVolumeTableActions/MigrateVolumeModal.tsx:109]: maybe consider concatenating volume.migration

**Review by SupaJoon — CHANGES_REQUESTED** [review]
looking for feedback/answer to my question on how the dot notation will be parsed later on

File: apps/spruce/src/analytics/spawn/useSpawnAnalytics.ts [hunk 2]
```diff
@@ -1,41 +1,39 @@
...
       name: "Created a spawn host";
-      isMigration: boolean;
-      params: Omit<
-        SpawnHostMutationVariables["spawnHostInput"],
-        "publicKey" | "userDataScript" | "setUpScript"
-      >;
+      "is.volume.migration": boolean;
+      "is.workstation": boolean;
+      "distro.id": string;
+      "no.expire": boolean;
```
[the remark this answers is not in the document]
Comment (khelif96) [reply apps/spruce/src/analytics/spawn/useSpawnAnalytics.ts:18]: I updated this alongside the rest of the params with the `host` prefix e.g `host.is.unexpirable`, `host.is.workstation` So it is easier to group by host attributes.

File: apps/spruce/src/pages/spawn/spawnVolume/spawnVolumeTableActions/EditVolumeModal.tsx [hunk 3]
```diff
@@ -67,7 +67,7 @@ export const EditVolumeModal: React.FC<Props> = ({
     const mutationInput = formToGql(initialState, formState, volume.id);
     spawnAnalytics.sendEvent({
       name: "Changed spawn volume settings",
-      params: mutationInput,
+      "volume.no.expire": mutationInput.noExpiration,
```
[the remark this answers is not in the document]
Comment (khelif96) [reply apps/spruce/src/pages/spawn/spawnVolume/spawnVolumeTableActions/EditVolumeModal.tsx:70]: Updated to `volume.is.unexpirable` for consistency with the host attributes

File: apps/spruce/src/analytics/patches/useProjectPatchesAnalytics.ts [hunk 4]
```diff
@@ -5,13 +5,13 @@ import { slugs } from "constants/routes";
...
+  | { name: "Changed project"; "project.identifier": string }
   | { name: "Clicked patch link" }
   | {
       name: "Filtered for patches";
-      filterBy: string;
-      includeHidden: boolean;
-      includeCommitQueue: boolean;
+      "filter.by": string;
+      "filter.include.hidden": boolean;
+      "filter.include.commit.queue": boolean;
```
[the remark this answers is not in the document]
Comment (khelif96) [reply apps/spruce/src/analytics/patches/useProjectPatchesAnalytics.ts]: This is the standard naming convention for otel attributes. I also updated some attributes to use namespaces as how they are described in the docs below
 https://opentelemetry.io/docs/specs/semconv/general/attribute-naming/

[the remark this answers is not in the document]
Comment (SupaJoon) [reply apps/spruce/src/analytics/patches/useProjectPatchesAnalytics.ts]: Neat, the doc states that namespaces are not meant to be hierarchical and are actually meant for preventing name clashes.

**Review by SupaJoon — APPROVED** [review]

**Pull request merged**